--- worker.sh.orig	2015-11-05 03:51:12.200224572 +0000
+++ worker.sh	2015-11-05 03:50:45.405474272 +0000
@@ -19,6 +19,24 @@
 
 set -e
 
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+# See if there's a different CILIUM_ROOT path set
+if [ -z ${CILIUM_ROOT} ]; then
+    CILIUM_ROOT=$( cd "$( dirname "${dir}/../../../../" )" && pwd )
+    echo "CILIUM_ROOT not set, using default: ${CILIUM_ROOT}"
+else
+    echo "CILIUM_ROOT is set to: ${CILIUM_ROOT}"
+fi
+
+# See if there's a different DOCKER_ENDPOINT set
+if [ -z ${DOCKER_ENDPOINT} ]; then
+    DOCKER_ENDPOINT="unix:///var/run/docker.sock"
+    echo "DOCKER_ENDPOINT not set, using default: ${DOCKER_ENDPOINT}"
+else
+    echo "Docker endpoint is set to: ${DOCKER_ENDPOINT}"
+fi
+
 # Make sure docker daemon is running
 if ( ! ps -ef | grep "/usr/bin/docker" | grep -v 'grep' &> /dev/null  ); then
     echo "Docker is not running on this machine!"
@@ -27,7 +45,7 @@
 
 # Make sure k8s version env is properly set
 if [ -z ${K8S_VERSION} ]; then
-    K8S_VERSION="1.0.3"
+    K8S_VERSION="1.0.7"
     echo "K8S_VERSION is not set, using default: ${K8S_VERSION}"
 else
     echo "k8s version is set to: ${K8S_VERSION}"
@@ -105,6 +123,12 @@
 
 # Start k8s components in containers
 start_k8s() {
+    if [ -f ${CILIUM_ROOT}/images/etcd.ditar ]; then
+        sudo docker -H unix:///var/run/docker-bootstrap.sock load -i ${CILIUM_ROOT}/images/etcd.ditar
+    fi
+    if [ -f ${CILIUM_ROOT}/images/flannel.ditar ]; then
+        sudo docker -H unix:///var/run/docker-bootstrap.sock load -i ${CILIUM_ROOT}/images/flannel.ditar
+    fi
     # Start flannel
     flannelCID=$(sudo docker -H unix:///var/run/docker-bootstrap.sock run -d --restart=always --net=host --privileged -v /dev/net:/dev/net quay.io/coreos/flannel:0.5.3 /opt/bin/flanneld --etcd-endpoints=http://${MASTER_IP}:4001 -iface="eth0")
 
@@ -152,6 +176,9 @@
 
     # sleep a little bit
     sleep 5
+
+    # Start cilium
+    $dir/../../../entrypoint.sh infect
     
     # Start kubelet & proxy in container
     # TODO: Use secure port for communication
@@ -172,7 +199,8 @@
         --v=2 --address=0.0.0.0 --enable-server \
         --cluster-dns=10.0.0.10 \
         --cluster-domain=cluster.local \
-        --containerized
+        --containerized \
+        --docker-endpoint=${DOCKER_ENDPOINT}
     
     docker run \
         -d \
