--- worker.sh.orig	2015-10-20 03:18:48.430731457 +0100
+++ worker.sh	2015-10-20 03:17:51.034266630 +0100
@@ -19,6 +19,22 @@
 
 set -e
 
+# See if there's a different DOCKER_ENDPOINT set
+if [ -z ${DOCKER_ENDPOINT} ]; then
+    DOCKER_ENDPOINT="unix:///var/run/docker.sock"
+    echo "DOCKER_ENDPOINT not set, using default: ${DOCKER_ENDPOINT}"
+else
+    echo "Docker endpoint is set to: ${DOCKER_ENDPOINT}"
+fi
+
+# See if there's a different IP set
+if [ -z ${IP} ]; then
+    IP=$(hostname -i)
+    echo "IP not set, using default: ${IP}"
+else
+    echo "IP is set to: ${IP}"
+fi
+
 # Make sure docker daemon is running
 if ( ! ps -ef | grep "/usr/bin/docker" | grep -v 'grep' &> /dev/null  ); then
     echo "Docker is not running on this machine!"
@@ -62,7 +78,7 @@
         *64)
             ;;
         *)
-	        echo "Error: We currently only support 64-bit platforms."	    
+	        echo "Error: We currently only support 64-bit platforms."
 	        exit 1
 	        ;;
     esac
@@ -147,7 +163,10 @@
 
     # sleep a little bit
     sleep 5
-    
+
+    # Start cilium
+    ../../../entrypoint.sh infect
+
     # Start kubelet & proxy in container
     docker run \
         --net=host \
@@ -162,10 +181,11 @@
         gcr.io/google_containers/hyperkube:v${K8S_VERSION} \
         /hyperkube kubelet --api-servers=http://${MASTER_IP}:8080 \
         --v=2 --address=0.0.0.0 --enable-server \
-        --hostname-override=$(hostname -i) \
+        --hostname-override="${IP}" \
         --cluster-dns=10.0.0.10 \
-        --cluster-domain=cluster.local
-    
+        --cluster-domain=cluster.local \
+	--docker-endpoint="${DOCKER_ENDPOINT}"
+
     docker run \
         -d \
         --net=host \
